docs/build.py

```diff
@@ -128,13 +153,29 @@ def clean_dist_dir():
 
     def generate_docs():
         template_ctx = dict(ftd_version=ftd_version, sample_dir=DEFAULT_SAMPLES_DIR)
-        ModelDocGenerator(DEFAULT_TEMPLATE_DIR, template_ctx, api_spec).generate_doc_files(args.dist, args.models)
-        OperationDocGenerator(DEFAULT_TEMPLATE_DIR, template_ctx, api_spec).generate_doc_files(args.dist, args.models)
-        ModuleDocGenerator(DEFAULT_TEMPLATE_DIR, template_ctx, DEFAULT_MODULE_DIR).generate_doc_files(args.dist)
-        StaticDocGenerator(STATIC_TEMPLATE_DIR, template_ctx).generate_doc_files(args.dist)
+        if args.doctype == DocType.ftd_ansible:
+            ModelDocGenerator(DEFAULT_TEMPLATE_DIR, template_ctx, api_spec).generate_doc_files(args.dist, args.models)
+            OperationDocGenerator(DEFAULT_TEMPLATE_DIR, template_ctx, api_spec).generate_doc_files(args.dist, args.models)
+            ModuleDocGenerator(DEFAULT_TEMPLATE_DIR, template_ctx, DEFAULT_MODULE_DIR).generate_doc_files(args.dist)
+            StaticDocGenerator(STATIC_TEMPLATE_DIR, template_ctx).generate_doc_files(args.dist)
+        elif args.doctype == DocType.ftd_api:
+            try:
+                errors_codes = api_client.fetch_error_codes()
+            except Exception as e:
```

Comment:
Please, narrow the exception class to `HTTPError` or smth more specific.

Reply:
in progress

Reply:
Why do we expect `JSONDecodeError` to be thrown here? If the document does not exist, it would be logical for the code to expect 4XX HTTP error.
If needed, let's update `FtdApiClient._send_request` to throw an explicit exception when the response code is not 2XX.

Reply:
cause in case document does not exist - server will return standard HTML page - which can't be parsed as json.